### ISSUE TITLE
Add Visual Regression Tests, Accessibility Tests and Code Coverage

### ISSUE THREAD

**Issue by ovflowd** [nodejs/nodejs.org#5278]
This issue is about adding 3 new kinds of testing for the new `major/website-redesign` branch:

- [ ] **Visual Regression Tests**
  - Allows us to catch when translation changes break the layout or when changes to components, global styles or et cetera break the layout
  - Should be done for each language
  - Should be done after we switch from the old layout to the new one, as we want to snapshot the new pages/routes
    - Should be done as we migrate each individual page
- [ ] **Code Coverage**
  - Allows us to measure how covered our codebase is with tests
  - Should ignore scripts, pages and styles
- [ ] **Accessibility Testing**
  - Allows us to ensure that we reached at least the minimum requirements for an accessible page for screen readers, colour schemes and font rendering
  - Should be done after we switch from the old layout to the new one, as we want to test the accessibility of the new layout
    - Should be done as we migrate each individual page

**Comment by Harkunwar** [comment]
`Visual Regression Tests` could be added using the Storybook, as well as for multiple languages. One concern I would like to mention is that Visual Regression tests do not work cross platform. The way the font renders is slightly different between Windows, Mac and Linux, which causes tests to fail when image snapshots are generated on one platform and tested on another. One way I've mitigated such issue before is using docker locally to generate snapshots, so it's always under a linux environment. The other option would be Github Action to have a manual way of directly updating the branch through our pipeline. Docker on mac doesn't have great performance. 

I can look into how `code coverage` could be added with `jest`. The `code coverage` addon for `Storybook` seems to be not working at the moment and I'm investigating that and might make a PR to resolve it soon.

`Accessibility Testing` can be integrated using Storybook as well: https://storybook.js.org/docs/react/writing-tests/accessibility-testing

**Comment by ovflowd** [comment]
> The other option would be Github Action to have a manual way of directly updating the branch through our pipeline. Docker on mac doesn't have great performance.

We can skip specific tests based on the matrix environment. Eg `if: ${{ matrix.os }} === 'ubuntu-latest'`. I think or something of sort

**Comment by ovflowd** [comment]
> I can look into how code coverage could be added with jest. The code coverage addon for Storybook seems to be not working at the moment and I'm investigating that and might make a PR to resolve it soon.

I guess using a GitHub Action Bot such as https://github.com/marketplace/actions/jest-coverage-report could use the output of `.coverage` folders. (Note since we have multiple steps with tests we might need to merge the reports, or just use the reports from the `test:unit:ci` step, which might be incomplete, but again, I doubt that just the regular Snapshot testing of a Component adds anything to the Coverage

**Comment by ovflowd** [comment]
> Accessibility Testing can be integrated using Storybook as well: [storybook.js.org/docs/react/writing-tests/accessibility-testing](https://storybook.js.org/docs/react/writing-tests/accessibility-testing)

Yeah, we could add this built-in to the test runner and maybe there's also a Bot for commenting this sort of things, not sure.

**Comment by ovflowd** [comment]
<sup>Off-topic: But @Harkunwar could you join the OpenJS Foundation Slack and give a look into this https://github.com/nodejs/nodejs.org/issues/5342 if you don't mind? :)</sup>

**Comment by ovflowd** [comment]
We've added Visual Regression Tests with Chromatic! We also have Code Coverage Bot Comment on PRs now. And we can update our Chromatic configuration to identify i18n issues.
